.github/workflows/manylinux_2_28.yml

```
@@ -447,9 +520,75 @@ jobs:
         run: python -m pytest -v ./${{ matrix.test.cmd }}
         working-directory: ${{ env.SRC_DIR }}
 
+  genai_nodejs_tests:
+    name: Node.js bindings tests
+    needs: [ smart_ci, openvino_download, genai_build_nodejs ]
+    if: ${{ fromJSON(needs.smart_ci.outputs.affected_components).JS_API }}
+    timeout-minutes: 20
+    defaults:
+      run:
+        shell: bash
+    runs-on: aks-linux-4-cores-16gb
+    container:
+      image: openvinogithubactions.azurecr.io/ov_test/ubuntu_22_04_x64:${{ needs.openvino_download.outputs.docker_tag }}
+      volumes:
+        - /mount:/mount
+        - ${{ github.workspace }}:${{ github.workspace }}
+      options: -e HF_TOKEN
+
+    env:
+      OV_INSTALL_DIR: ${{ github.workspace }}/ov
+      SRC_DIR: ${{ github.workspace }}/src
+      NODE_VERSION: 22
+
+    steps:
+      - name: Clone openvino.genai
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          path: ${{ env.SRC_DIR }}
+          submodules: recursive
```

Please add
`TARGET_BRANCH: ${{ inputs.target-branch || github.base_ref || github.event.merge_group.base_ref || github.ref }}` to the env block for the whole workflow and use it here like this:
```suggestion
          submodules: recursive
          ref: ${{ env.TARGET_BRANCH }}
```
This is for the ability to build package manually (not via PR trigger) for the given branch if needed